src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php

```diff
@@ -73,17 +73,25 @@ public function redirectAction(Request $request)
      */
     protected function resolveRedirectUrl($redirectUrl, $requestUri)
     {
-        $redirectInfo = $this->parseUrl($redirectUrl);
         $requestInfo = $this->parseUrl($requestUri);
 
-        $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+        $url = sprintf('%s://%s', $requestInfo['scheme'], $requestInfo['host']);
 
         if (isset($requestInfo['port'])) {
             $url .= ':' . $requestInfo['port'];
         }
 
-        if (isset($redirectInfo['path'])) {
-            $url .= $redirectInfo['path'];
+        if ($redirectUrl) {
+            $redirectInfo = $this->parseUrl($redirectUrl);
+            $url = sprintf('%s://%s', $requestInfo['scheme'], $redirectInfo['host']);
+
+            if (isset($requestInfo['port'])) {
+                $url .= ':' . $requestInfo['port'];
+            }
+
+            if (isset($redirectInfo['path'])) {
+                $url .= $redirectInfo['path'];
+            }
         }
 
         if (isset($requestInfo['path'])) {
```

Comment:
in the case of the slash redirect the redirectUrl is null(because domain + language is set correctly)  if it would not be null it would build a false url `sulu.lo/de/de/....`